kratos/geometries/quadrilateral_3d_4.h

```diff
@@ -784,41 +776,33 @@ template<class TPointType> class Quadrilateral3D4
      * @see DeterminantOfJacobian
      * @see InverseOfJacobian
      */
-    Matrix& Jacobian( Matrix& rResult,
-                              IndexType IntegrationPointIndex,
-                              IntegrationMethod ThisMethod ) const override
+    Matrix& Jacobian( 
+        Matrix& rResult,
+        IndexType IntegrationPointIndex,
+        IntegrationMethod ThisMethod 
+        ) const override
     {
-        //setting up size of jacobian matrix
+        // Setting up size of jacobian matrix
         rResult.resize( 3, 2, false );
-        //derivatives of shape functions
-        ShapeFunctionsGradientsType shape_functions_gradients =
-            CalculateShapeFunctionsIntegrationPointsLocalGradients( ThisMethod );
-        Matrix ShapeFunctionsGradientInIntegrationPoint =
-            shape_functions_gradients( IntegrationPointIndex );
-        //values of shape functions in integration points
-//         Vector ShapeFunctionsValuesInIntegrationPoint = ZeroVector( 4 );
-//         /*vector<double>*/
-//         ShapeFunctionsValuesInIntegrationPoint = row(
-//                 CalculateShapeFunctionsIntegrationPointsValues( ThisMethod ),
-//                 IntegrationPointIndex );
-
+        // Derivatives of shape functions
+        ShapeFunctionsGradientsType shape_functions_gradients = CalculateShapeFunctionsIntegrationPointsLocalGradients( ThisMethod );
+        
         //Elements of jacobian matrix (e.g. J(1,1) = dX1/dXi1)
         //loop over all nodes
-
         for ( unsigned int i = 0; i < this->PointsNumber(); i++ )
         {
             rResult( 0, 0 ) +=
-                ( this->GetPoint( i ).X() ) * ( ShapeFunctionsGradientInIntegrationPoint( i, 0 ) );
+                ( this->GetPoint( i ).X() ) * ( shape_functions_gradients[IntegrationPointIndex]( i, 0 ) );

```

Comment:
It is not needed if you get the right reference

Reply:
Ok